linode/manager
# feat: [UIE-8138] - IAM RBAC: add new no assigned roles component

## Description 📝

IAM RBAC - new no assigned roles component

## Changes  🔄

List any change(s) relevant to the reviewer.

- new no assigned roles component

## Target release date 🗓️

1/14/25 (dev)

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| 📷 |![image](https://github.com/user-attachments/assets/1241987e-4a8d-4e0d-ad8c-8d2cd578e331)|
| 📷 |![image](https://github.com/user-attachments/assets/c16469ad-a0fa-401a-b9dc-7a840e763fa9)|
| 📷 |![image](https://github.com/user-attachments/assets/f7e4fa17-3852-44eb-9d15-bf7ef7f129ac)|
| 📷 |![image](https://github.com/user-attachments/assets/d60464b8-4f5e-4513-a83b-4fbdeeabc136)|

## How to test 🧪

### Prerequisites

(How to setup test environment)

- Ensure the Identity and Access Beta flag is enabled in dev tools
- Click on the username in the users table and go to the tab `Assigned Roles` or click on the user's menu `View User Roles`
- Click on the username in the users table and go to the tab `Assigned Resources` or click on the user's menu `View User Roles` and go to the tab `Assigned Resources`
- Only mock user has some roles assigned


### Verification steps


- [x]  Confirm no assigned roles component renders for `Assigned Roles` tab when the user doesn't have any roles assigned.
- [x]  Confirm no assigned roles component doesn't renders for `Assigned Roles` tab when the user has some roles assigned.
- [x] Confirm no assigned roles component renders for `Assigned Resources` tab when the user doesn't have any roles assigned.
- [x]  Confirm no assigned roles component doesn't renders for `Assigned Resources` tab when the user has some roles assigned.


<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
